### ISSUE TITLE
Tiled fails to render alpha component of tintcolor correctly for some tiles

### ISSUE THREAD

**Issue by onpon4** [mapeditor/tiled#2974]
I have an example in my game engine which uses a Tiled map to design a level. As I made some changes to test the addition of tintcolor support to the engine, I noticed something odd: I had use the tint color to adjust alpha transparency simply because the images used were black (and thus wouldn't be affected by the color component of tint color, but it seems Tiled isn't actually correctly applying the blending correctly to alpha transparency in all cases. At first I thought this might be #2930, so I changed the color of the tiles from black to gray and applied a color tint, confirming that this is a different issue.

Here is the Tiled JSON file (as the JSON format is what I use) and the tileset file and tileset image it uses:

[bad_tiled.zip](https://github.com/mapeditor/tiled/files/5816015/bad_tiled.zip)

Here is a screenshot demonstrating what it looks like in the test case for the engine, and what it should look like:

![Screenshot from 2021-01-14 12-28-30](https://user-images.githubusercontent.com/2217995/104626948-6ec42a00-5664-11eb-8456-a57315bfbf79.png)

But in Tiled 1.4.3, it looks like this:

![Screenshot from 2021-01-14 12-32-14](https://user-images.githubusercontent.com/2217995/104627106-9adfab00-5664-11eb-8b26-7d5a3f2c6334.png)

The image exported by Tiled is quite different, and this may perhaps give some insight into the cause of the problem:

![level](https://user-images.githubusercontent.com/2217995/104627325-e3976400-5664-11eb-9ba7-19c7507ef5d9.png)

I'm not sure how easy this is to see in a Web browser, but if you open it in an image viewer that shows transparent sections as a checkerboard pattern, you can clearly see that the parts with the gray square tiles are seemingly erasing the cyan background color.

Tested via the 1.4.3 AppImage on Debian 10 "Buster".

**Comment by Kiamo2** [comment]
Strangely his example in 'bad_tiled.zip' now works.

But in other cases it still happens that reducing alpha of the tint color produces wrong results.
The color is way to dark and with alpha = 0 it results in a solid black.
It seems that a black instead of e.g. a transparent background color is assumed.
And it seems that this happes only if the tile image does not show the transparent background somewhere i.e. the background is completely concealed.
If a tranparent background in the tile image somewhere shines through the tinting appears correct.

**Comment by benryhenson** [comment]
I see this exact behavior (darkening instead of transparency) in shaders and in game engines like Unity etc where a texture lacks an alpha channel. In those cases, multiplying by alpha results in the RBG values lowering, hence darkening. If this is the source of the issue, the question is if the file itself has the problem of lacking an alpha channel, or if the issue is on Tiled's side (or QT) when importing/rendering the png. I'm supposing the issue is on the rendering side, since if a TileLayer's opacity is <100%, then everything renders correctly.

**Comment by Kiamo2** [comment]
Really? I do not observe correct rendering with a TileLayer's opacity < 100%.
In my tests the effect (way too dark) is slightly less heavy but still there.

I'm a bit afraid it's on the Qt side - that they did s.th. but not enough.
Cause bad_tiled.zip suddenly works while supposedly in tiled nobody has done anything about this issue.

**Comment by benryhenson** [comment]
Yeah, I sent a video in the discord but here's a quick image. With layer tint set with a non-maximum alpha (in this case R122 G0 B0 A128), setting layer opacity <100 seems to resolve the issue. Which is of course redundant to tint alpha.

<img width="1524" height="805" alt="Image" src="https://github.com/user-attachments/assets/4aeb89d8-c074-4038-8642-8e2076c73e6e" />
<img width="1520" height="779" alt="Image" src="https://github.com/user-attachments/assets/f321ff29-b0f2-45b5-81ea-5e895ffdc74a" />

**Comment by benryhenson** [comment]
Ok I did some research and tinkering and despite me making 1000% sure my Inkscape export setting is RGBA, the file still comes out missing alpha channel. So the good news is I think Tiled/QT are "working as intended" and it is in fact my image file that is problematic.

<img width="938" height="217" alt="Image" src="https://github.com/user-attachments/assets/b9181ae2-db7b-4669-b688-12ab80a595d2" />

**Comment by Kiamo2** [comment]
My latest theory is now that the alpha channel of the image is left out or removed if "not needed" i.e. no transparent background is anywhere shining through. If transparent background shines through (a single pixel is enough) the tinting with that image is well behaving.
I suspect it's left out or removed in the exported image despite all settings and you can do nothing about it.
Maybe even libpng automatically does that...?
